Azure/sonic-buildimage
# [test] Adding Broadcom configuration file test

**- What I did**
In order to allow SONiC community to check in Broadcom configuration
file directly under device folder. We need to add this test to make
sure the contents of the configuration is compliant with Broadcom
specifications.

Added one config.bcm for Arista7260cx3 as example and also for positive test.

**- How to verify it**
With the change as-is, build passes, the built image executed on the target platform, the configuration was applied properly.

With following contents in following file:

localadmin@yingxie-vm0:~/src/sonic-buffers$ cat device/arista/x86_64-arista_7260cx3_64/Arista-7260CX3-C64/test-must-fail.config.bcm 
invalid_entry=invalid_value

Build hit following failure:

/sonic/src/sonic-device-data/tests /sonic/src/sonic-device-data/src
./config_checker ../../../device/arista/x86_64-arista_7260cx3_64/Arista-7260CX3-D108C8/th2-a7260cx3-64-112x50G+8x100G.config.bcm
Result: ../../../device/arista/x86_64-arista_7260cx3_64/Arista-7260CX3-D108C8/th2-a7260cx3-64-112x50G+8x100G.config.bcm PASSED the config check!
./config_checker ../../../device/arista/x86_64-arista_7260cx3_64/Arista-7260CX3-C64/test-must-fail.config.bcm
[line 1] Error: invalid_entry is not permitted
Result: ../../../device/arista/x86_64-arista_7260cx3_64/Arista-7260CX3-C64/test-must-fail.config.bcm FAILED the config check!
Makefile:6: recipe for target 'test' failed
make[3]: *** [test] Error 255
make[3]: Leaving directory '/sonic/src/sonic-device-data/src'
dh_auto_build: make -j1 returned exit code 2
debian/rules:6: recipe for target 'build' failed
make[2]: *** [build] Error 2
make[2]: Leaving directory '/sonic/src/sonic-device-data/src'
dpkg-buildpackage: error: debian/rules build gave error exit status 2
Makefile:8: recipe for target '/sonic/target/debs/sonic-device-data_1.0-1_all.deb' failed
make[1]: *** [/sonic/target/debs/sonic-device-data_1.0-1_all.deb] Error 2
make[1]: Leaving directory '/sonic/src/sonic-device-data'
[  FAIL LOG END  ] [ target/debs/sonic-device-data_1.0-1_all.deb ]
slave.mk:194: recipe for target 'target/debs/sonic-device-data_1.0-1_all.deb' failed
make: *** [target/debs/sonic-device-data_1.0-1_all.deb] Error 1
Makefile:87: recipe for target 'target/debs/sonic-device-data_1.0-1_all.deb' failed
make: *** [target/debs/sonic-device-data_1.0-1_all.deb] Error 2
